docs/coordinates/inplace.rst

```diff
@@ -0,0 +1,50 @@
+In Place Modification of Coordinates
+====================================
+
+Coordinates are generally considered to be immutable. If you want to create
+another coordinate frame with different data you should use
+`~astropy.coordinates.BaseCoordinateFrame.realize_frame`, this is the safest way
+to change the data in a frame as it creates a new frame with that data.
+Creating a new frame can be relatively slow, however, particularly for scalar
+coordinates. Hence some situations may require that data by changed in-place in
+an already existing frame object. This modification can be done by
+modifiying the values of the representation data as follows::
+
+    >>> import astropy.units as u
+    >>> from astropy.coordinates import SkyCoord
+    >>> c = SkyCoord([1,2],[3,4], unit='deg')
+    >>> c.data.lon[()] = [10, 20] * u.deg
+    >>> c
+    <SkyCoord (ICRS): (ra, dec) in deg
+        [( 10.,  3.), ( 20.,  4.)]>
+
+
+This changes the longitude values of the frame. Unfortunately, doing just this
+introduces problems: `~astropy.coordinates.SkyCoord` and
+`~astropy.coordinates.BaseCoordinateFrame` cache various kinds of information to
+speed up some repeated operations. So we need to tell the cache that it should
+be cleared so that it can be re-calculated from the new data. This can be
+achieved by doing::
+
+    >>> c.cache.clear()
+
+or::
```

Comment:
I'm very happy with the `c.cache.clear()` way of clearing the cache. I think it should be immediately obvious that this works for frames as well, so would suggest removing the alternative (one right way to do it!).